modules/contact/src/ReferenceResidualProblem.C

```diff
@@ -115,6 +161,86 @@ ReferenceResidualProblem::initialSetup()
       mooseError("Could not find variable '", _refResidVarNames[i], "' in auxiliary system");
   }
 
+  unsigned int ungroup_index = 0;
+  if (_group_variables)
+    ungroup_index = (*_group_variables).size();
+
+  for (unsigned int i = 0; i < _solnVars.size(); ++i)
+  {
+    bool find_group = false;
+    if (_group_variables)
+    {
+      for (unsigned int j = 0; j < (*_group_variables).size(); ++j)
+      {
+        if (std::find((*_group_variables)[j].begin(),
+                      (*_group_variables)[j].end(),
+                      s.variable_name(_solnVars[i])) != (*_group_variables)[j].end())
+        {
+          _variable_group_num_index[i] = j;
+          find_group = true;
+          break;
+        }
+      }
+      if (!find_group)
+      {
+        _variable_group_num_index[i] = ungroup_index;
+        ungroup_index++;
+      }
+    }
+    else
+    {
+      _variable_group_num_index[i] = i;
+    }
+  }
+
+  if (_group_variables)
+  {
+    for (unsigned int i = 0; i < (*_group_variables).size(); ++i)
+    {
+      bool same_variable = true;
+      if ((*_group_variables)[i].size() > 1)
+      {
+        for (unsigned int j = 0; j < (*_group_variables)[i].size(); ++j)
+        {
+          for (unsigned int var_num = 0; var_num < s.n_vars(); var_num++)
+          {
+            if ((*_group_variables)[i][j] == s.variable_name(var_num))
+            {
+              if (j == 0)
+                same_variable = nonlinear_sys.isScalarVariable(_solnVars[var_num]);
+              else
+                same_variable =
+                    (same_variable == nonlinear_sys.isScalarVariable(_solnVars[var_num]) ? true
+                                                                                         : false);
+              break;
+            }
+          }
+        }
+      }
+      if (!same_variable)
+        mooseWarning("For group_variables parameter, Row ",
+                     i,
+                     " tries to group standard variables with scalar variables.");
```

Comment:
"In the 'group_variables' parameter, standard variables and scalar variables are grouped together in group ", i